src/tests/Interop/COM/Activator/Program.cs

```diff
@@ -259,11 +279,23 @@ static int Main(string[] doNotUse)
         {
             try
             {
+                bool builtInComDisabled = false;
+                var comConfig = AppContext.GetData("System.Runtime.InteropServices.BuiltInComInterop.IsSupported");
+                if (comConfig != null && !bool.Parse(comConfig.ToString()))
+                {
+                    builtInComDisabled = true;
+                }
+                Console.WriteLine($"Built-in COM Disabled?: {builtInComDisabled}");
+
                 InvalidInterfaceRequest();
-                ClassNotRegistered();
-                NonrootedAssemblyPath();
-                ValidateAssemblyIsolation();
-                ValidateUserDefinedRegistrationCallbacks();
+                ClassNotRegistered(builtInComDisabled);
+                NonrootedAssemblyPath(builtInComDisabled);
+                ValidateAssemblyIsolation(builtInComDisabled);
+                if (!builtInComDisabled)
+                {
+                    // We don't test with builtInComDisabled since ValidateAssemblyIsolation() above covers this scenarios
```

Comment:
```suggestion
                    // We don't test this scenario with builtInComDisabled since it is covered by ValidateAssemblyIsolation() above
```